src/client.rs

```diff
@@ -85,6 +95,8 @@ fn client_loop(config: ValidatorConfig) {
             }
         } else {
             info!("[{}@{}] session closed gracefully", chain_id, &addr);
+            // Indicate to the outer thread it's time to terminate
+            Arc::clone(should_term).swap(true, Ordering::Relaxed);
```

Comment:
Same as above. Can't we just do:
```suggestion
            should_term.swap(true, Ordering::Relaxed);
```

Reply:
Also fixed now.